data/trie/branchNode.go

```diff
@@ -778,3 +778,18 @@ func (bn *branchNode) getAllHashes(db data.DBWriteCacher) ([][]byte, error) {
 
 	return hashes, nil
 }
+
+func (bn *branchNode) verifyProof(key []byte) (bool, []byte, []byte) {
```

Comment:
Maybe rename this to getEncodedBytesAndHash? as this method does not verify the proof. Valid on all 3 types of nodes.

Reply:
Renamed to getNextHashAndKey.